examples/ags/benchmark/humaneval.py

```diff
@@ -0,0 +1,171 @@
+    case = get_human_eval_plus()[f"{id}"]
+    if mode == "ags":
+        solution_result = await solver(case['prompt'],ensemble_count=5)
+        sample_dict = dict(task_id=case['task_id'], solution=solution_result['final_solution'])
+    elif mode == "alpha":
+        solution_result = await solver.alpha_codium(case['task_id'], case['prompt'], ensemble_count=5)
+        sample_dict = dict(task_id=case['task_id'], solution=solution_result['final_solution'])
+    elif mode == "llm":
+        solution_result =  await generate_code_block(case['prompt'],case['entry_point'])
+        sample_dict = dict(task_id=case['task_id'], solution=solution_result['code_solution'])
+        print(sample_dict)
+    with open(result_path, mode='a') as f:
+        f.write(json.dumps(sample_dict) + '\n')
+    jsonl_ranker(result_path, result_path)
+
+async def samples_generate(mode:str, result_path:str="samples.jsonl"):
+    cases = list(get_human_eval_plus().values())
+    file_lock = asyncio.Lock()
+    
+    async def solve_and_write(case, mode):
+        try:
+            if mode == 'llm':
+                solution_result = await generate_code_block(problem_description=case['prompt'], function_name=case['entry_point'])
+                # solution_result = await generate_code(case['prompt'])
+                sample_dict = {
+                'task_id': case['task_id'],
+                'solution': solution_result['code_solution']
+                }
+            elif mode == "ags":
+                solution_result = await solver(case['prompt'], ensemble_count=5)
+                sample_dict = {
+                'task_id': case['task_id'],
+                'solution': solution_result['final_solution']
+                }
+            elif mode == "alpha":
+                solution_result = await solver.alpha_codium(case['task_id'], case['prompt'], ensemble_count=5)
+                sample_dict = {
+                'task_id': case['task_id'],
+                'solution': solution_result['final_solution']
+                }
+            # TODO 解决  final_solution 问题之后就可以开始正式测评了
+            async with file_lock:
+                async with aiofiles.open(result_path, mode='a') as f:
+                    await f.write(json.dumps(sample_dict) + '\n')
+            return None
+
+        except Exception as e: 
+            print(e)
+            return case['task_id']
+
+    tasks = [solve_and_write(case, mode) for case in cases]
+    results = await asyncio.gather(*tasks)
+    failed_tasks = [task_id for task_id in results if task_id is not None]
+
+    if failed_tasks:
+        print(failed_tasks)
+        if mode == 'llm':
+            for task_id in failed_tasks:
+                case = get_human_eval_plus()[task_id]
+                for _ in range(3):
+                    try:
+                        solution_result = await generate_code_block(case['prompt'],function_name=case['entry_point'])
+                        task_dict = {
```

Comment:
refs https://docs.deepwisdom.ai/main/en/guide/contribute/contribute_guide.html#before-submission to format code